lib/ex_machina/ecto_strategy.ex

```diff
@@ -29,36 +29,25 @@ defmodule ExMachina.EctoStrategy do
   defp cast(record) do
     record
     |> cast_all_fields
+    |> cast_all_embeds
     |> cast_all_assocs
   end
 
-  defp cast_all_fields(struct) do
-    struct
-    |> ExMachina.Ecto.drop_ecto_fields
-    |> Map.keys
-    |> cast_all_fields(struct)
-  end
+  defp cast_all_fields(%{__struct__: schema} = struct) do
+    field_keys = schema_fields(schema)
+
+    Enum.reduce(field_keys, struct, fn(field_key, struct) ->
```

Comment:
it doesn't look like you use `field_keys` anywhere else, so stylistically you could do this:

```
schema
|> schema_fields
|> Enum.reduce(struct, fn(field_key, struct) -> 
...
```